### ISSUE TITLE
Idea: add a public `render` method to Usage Drivers

### ISSUE THREAD

**Issue by thermokarst** [qiime2/qiime2#562]
**Improvement Description**
A few of the existing drivers out there are starting to congeal around having a unified `render` method, which translates a hydrated `Scope` to relevant interface-specific invocations. By making this part of the public API we can start to rely on this method (as in, we expect drivers to implement the `render` method).

**Current Behavior**
No `Usage.render` method.

**Proposed Behavior**
`Usage.render` would return a generator that allows the caller to iterate over relevant interface-specific outputs/invocations.

**Comment by ebolyen** [comment]
Would drivers have a mechanism to indicate that they aren't renderable? I think there are other kinds of Usages out there.

**Comment by thermokarst** [comment]
Yeah, good question - I think it would be a bit like [`_assert_has_line_matching_`](https://github.com/qiime2/qiime2/blob/66ea209e136fc2e1a69268cdba81d02e1130d137/qiime2/sdk/usage.py#L682) - its an optional method, but for drivers that have a need for rendering it would provide some consistency.
